adapters/p2p2consensus/transaction.go

```diff
@@ -5,32 +5,32 @@ import (
 
 	"github.com/NethermindEth/juno/adapters/p2p2core"
 	"github.com/NethermindEth/juno/core"
-	"github.com/NethermindEth/juno/utils"
 	p2pconsensus "github.com/starknet-io/starknet-p2pspecs/p2p/proto/consensus/consensus"
 )
 
-func AdaptTransaction(t *p2pconsensus.ConsensusTransaction, network *utils.Network) (core.Transaction, core.Class) {
+func AdaptTransaction(t *p2pconsensus.ConsensusTransaction) (core.Transaction, core.Class, error) {
 	if t == nil {
-		return nil, nil
+		return nil, nil, nil
+	}
+	if err := validateConsensusTransaction(t); err != nil {
+		return nil, nil, err
 	}
-
 	switch t.Txn.(type) {
 	case *p2pconsensus.ConsensusTransaction_DeclareV3:
 		tx := t.GetDeclareV3()
 		// Todo: we pass in CompiledClassHash here, but in sync we pass in ClassHash.
 		// Are we expected to compile the class hash here???
 		class := p2p2core.AdaptCairo1Class(tx.Class)
-		return p2p2core.AdaptDeclareV3TxnCommon(tx.Common, tx.Common.CompiledClassHash, t.TransactionHash), &class
-
+		return p2p2core.AdaptDeclareV3TxnCommon(tx.Common, tx.Common.CompiledClassHash, t.TransactionHash), &class, nil
```

Comment:
We need a validate function for each transaction type, called inside each of these cases.